static/js/integrations_dev_panel.js

```diff
@@ -62,16 +63,60 @@ function get_selected_integration_name() {
     return $("#integration_name").children("option:selected").val();
 }
 
+function get_fixture_format(fixture_name) {
+    var pieces = fixture_name.split(".");
+    return pieces[pieces.length - 1];
+}
+
+function get_custom_http_headers() {
+    var custom_headers = $("#custom_http_headers").val();
+    if (custom_headers !== "") {
+        // JSON.parse("") would trigger an error, as empty strings do not qualify as JSON.
+        try {
+            // Let JavaScript validate the JSON for us.
+            custom_headers = JSON.stringify(JSON.parse(custom_headers));
+        } catch (err) {
+            set_message("Custom HTTP headers are not in a valid JSON format.", "warning");
+            return;
+        }
+    }
+    return custom_headers;
+}
+
+function set_results(response) {
+    /* After sending a webhook fixture message, the backend will let us know the exact response
+    information for each of the responses sent (which is just 1 for send, but it may be (and
+    usually is) more than 1 for send all)
+
+    NOTE: This following bit of code is not efficient and might need to be optimized.
+          Since this is a devtool and the number of test fixtures for any given
+          integration is pretty low, optimization here is not a high priority task. */
+    var responses = response.responses;
+
+    var data = "Results:\n\n";
+    responses.forEach(function (response) {
+        if (response.fixture_name !== undefined) {
+            data += "Fixture:            " + response.fixture_name;
+            data += "\nStatus Code:    "  + response.status_code;
+        } else {
+            data += "Status Code:    "  + response.status_code;
+        }
+        data += "\nResponse:       " + response.message + "\n\n";
+    });
+    $("#idp-results")[0].value = data;
+}
+
 function load_fixture_body(fixture_name) {
     /* Given a fixture name, use the loaded_fixtures dictionary to set the fixture body field. */
     var integration_name = get_selected_integration_name();
-    var element = loaded_fixtures[integration_name][fixture_name];
-    var fixture_body = JSON.stringify(element, null, 4); // 4 is for the pretty print indent factor.
-
+    var fixture_body = loaded_fixtures[integration_name][fixture_name];
     if (fixture_body === undefined) {
         set_message("Fixture does not have a body.", "warning");
         return;
     }
+    if (get_fixture_format(fixture_name) === "json") {
+        fixture_body = JSON.stringify(fixture_body, null, 4);// 4 is for pretty print .
```

Comment:
Though oops, missed this one; can you fix it?  Also the ` .` is weird.

Reply:
It looks like you already fixed it :slightly_smiling_face: